jsm.go

```diff
@@ -935,8 +935,8 @@ func (js *js) StreamNames(ctx context.Context) <-chan string {
 	return ch
 }
 
-func getJSContextOpts(defs *js, opts ...JSOpt) (*js, context.CancelFunc, error) {
-	var o js
+func getJSContextOpts(defs *js, opts ...JSOpt) (*jsOpts, context.CancelFunc, error) {
```

Comment:
could just pass the options here?

```suggestion
func getJSContextOpts(defs *jsOpts, opts ...JSOpt) (*jsOpts, context.CancelFunc, error) {
```